### PR TITLE
Finalize adding todont form

### PR DESCRIPTION
**Pull request by TitoOn**
close #48

### CODE REVIEW

File: src/main/java/ch/hearc/todont/controllers/CreateToDontController.java [hunk 1]
```diff
@@ -40,18 +44,22 @@ public String get(Model model) {
...
-     * @param user      User adding the to-dont
+     * @param userDetail User adding the to-dont
      * @param newToDont Returned value from form
      * @return The page template name
      */
     @PostMapping("/create")
     public String createToDont(@RequestParam(name = "usernames[]") String[] usernames, 
         @RequestParam(name = "moderatorBoolean[]") String[] moderators,
-        @AuthenticationPrincipal User user,
+        @AuthenticationPrincipal org.springframework.security.core.userdetails.User userDetail,
```
Comment (g-wizzy) [inline src/main/java/ch/hearc/todont/controllers/CreateToDontController.java:54]: Could you please copy this to all the mappings that use the User's info ? Along with the line 57

File: src/main/java/ch/hearc/todont/models/Pledge.java [hunk 2]
```diff
@@ -114,4 +117,35 @@ public Timestamp getDateFailed() {
     public void setDateFailed(Timestamp dateFailed) {
         this.dateFailed = dateFailed;
     }
+
+    @Override
```
Comment (g-wizzy) [inline src/main/java/ch/hearc/todont/models/Pledge.java]: From here on, those methods are useless.

File: src/main/java/ch/hearc/todont/models/PledgeKey.java [hunk 3]
```diff
@@ -68,4 +67,9 @@ public boolean equals(Object obj) {
         }
         return true;
     }
+
+    @Override
```
Comment (g-wizzy) [inline src/main/java/ch/hearc/todont/models/PledgeKey.java]: Could be removed.

**Review by g-wizzy — CHANGES_REQUESTED** [review]

**Review by g-wizzy — APPROVED** [review]

**Review by Blatoy — APPROVED** [review]

**Comment by g-wizzy** [comment]
The User acquisition in the controllers sure is an ugly fix. Time is of the essence right now, and a proper solution will be implemented after the live demo.

**Pull request merged**